xtask/src/run.rs

```diff
@@ -206,31 +198,17 @@ pub fn run(opts: Options) -> Result<()> {
                 .try_exists()
                 .context("failed to check existence of gen_init_cpio")?
             {
-                // TODO(https://github.com/oxidecomputer/third-party-api-clients/issues/96): Use ETag-based caching.
-                let client = octorust::Client::new(
-                    String::from("aya-xtask-integration-test-run"),
-                    github_api_token.map(octorust::auth::Credentials::Token),
-                )?;
-                let octorust::Response {
-                    status: _,
-                    headers: _,
-                    body: octorust::types::ContentFile { mut content, .. },
-                } = tokio::runtime::Builder::new_current_thread()
+                let content = tokio::runtime::Builder::new_current_thread()
                     .enable_all()
                     .build()
                     .unwrap()
-                    .block_on(client.repos().get_content_file(
-                        "torvalds",
-                        "linux",
-                        "usr/gen_init_cpio.c",
-                        "master",
-                    ))
+                    .block_on(async {
+                        reqwest::get("https://github.com/torvalds/linux/raw/refs/heads/master/usr/gen_init_cpio.c")
```

Comment:
You'll want to use https://docs.rs/reqwest/latest/reqwest/blocking/struct.Client.html
so we can set headers.

[This doc](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28) suggests it would be wise to:

1. Set the header for the expect API version
2. Set the auth header if `--github-api-token` was provided

Furthermore, I would suggest using the same API endpoint as `octorust` was using due to the aforementioned issues with rate limiting.

https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28

You'll then want to parse the reponse with [`serde_json`](https://docs.rs/serde_json/latest/serde_json/#operating-on-untyped-json-values)